src/login_param.rs

```diff
@@ -133,14 +135,30 @@ pub struct LoginParam {
     pub addr: String,
     pub imap: ServerLoginParam,
     pub smtp: ServerLoginParam,
-    pub server_flags: i32,
     pub provider: Option<&'static Provider>,
     pub socks5_config: Option<Socks5Config>,
 }
 
 impl LoginParam {
     /// Load entered (candidate) account settings
     pub async fn load_candidate_params(context: &Context) -> Result<Self> {
+        let mut param = Self::load_invalid_candidate_params(context).await?;
+        ensure!(!param.addr.is_empty(), "Missing email address.");
+
+        // Only check for IMAP password, SMTP password is an "advanced" setting.
+        ensure!(!param.imap.password.is_empty(), "Missing (IMAP) password.");
+        if param.smtp.password.is_empty() {
+            param.smtp.password = param.imap.password.clone()
+        }
+        Ok(param)
+    }
+
+    /// Load entered (candidate) account settings without validation.
+    ///
+    /// This will result in a potentially invalid [`LoginParam`] struct as the values are
+    /// not validated.  Only use this if you want to show this directly to the user e.g. in
+    /// [`Context::get_info`].
+    pub async fn load_invalid_candidate_params(context: &Context) -> Result<Self> {
```

Comment:
I would rather name it `load_invalid_candidate_params_unchecked`, similar to standard `from_utf8_unchecked`. With the current name at first I thought it is some test function that provides invalid parameters for testing of validation code.

Reply:
that's a great suggestion indeed, thanks!